File: problems/HackerRank/dynamicArray.ts
function dynamicArray(n: number, queries: number[][]): number[] {
  const arr = Array.from({ length: n }, () => []) as any;

  const ans = [];
  let lastAnswer = 0;

  for (let i = 0; i < queries.length; i++) {
    let x = Number(queries[i][1]);
    let y = Number(queries[i][2]);
    let idx = (x ^ lastAnswer) % n;

    if (Number(queries[i][0]) === 1) {
      arr[idx].push(y);
    } else if (Number(queries[i][0]) === 2) {
      lastAnswer = arr[idx][y % arr[idx].length];
      ans.push(lastAnswer);
    }
  }
  return ans;
}

dynamicArray(3, [[12], [3]]);

File: problems/HackerRank/week5/dynamicArray.py
def dynamicArray(n, queries):
    arr = [[] for _ in range(n)]
    lastAnswer = 0
    ans = []
    
    for i in range(len(queries)):
        
        x = int(queries[i][1])
        y = int(queries[i][2])
        idx = (x ^ lastAnswer) % n
        
        if (int(queries[i][0]) == 1):
            arr[idx].append(y)
        elif (int(queries[i][0]) == 2):
            lastAnswer = arr[idx][y % len(arr[idx])]
            ans.append(lastAnswer)
            
    return ans